inspire_matcher/api.py

```diff
@@ -58,6 +59,8 @@ def match(record, config=None):
         current_app.logger.debug('No configuration provided. Falling back to the default configuration.')
         config = current_app.config['MATCHER_DEFAULT_CONFIGURATION']
 
+    config['index'] = get_prefixed_index(config['index'])
```

Comment:
here you're mutating the config you're receiving, which causes problems when you call it twice with the same config, and requires workarounds in the function. Better put it on L64 to avoid mutating the config (you can also remove support for ES <7, we're not using it any longer).